File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/Commons/RoleModel.cs

using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Model.Commons;
using VecompSoftware.DocSuiteWeb.Model.Entities.Tenants;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.Commons
{
    public class RoleModel
    {
        #region [ Constructors ]

        public RoleModel()
        {
            Children = new HashSet<RoleModel>();
        }

        public RoleModel(short? id) : this()
        {
            IdRole = id;
            EntityShortId = id;
        }

        #endregion

        #region [ Properties ]
        public short? EntityShortId { get; set; }
        public short? IdRole { get; set; }
        public Guid? UniqueId { get; set; }
        public string Name { get; set; }
        public string FullIncrementalPath { get; set; }
        public bool IsActive { get; set; }
        public short? IdRoleFather { get; set; }
        public string ServiceCode { get; set; }
        public RoleTypology RoleTypology { get; set; }
        public bool Collapsed { get; set; }
        public string EMailAddress { get; set; }
        public string RegistrationUser { get; set; }
        public DateTimeOffset RegistrationDate { get; set; }
        public string LastChangedUser { get; set; }
        public DateTimeOffset? LastChangedDate { get; set; }

        public string RoleLabel { get; set; }
        public AuthorizationRoleType AuthorizationType { get; set; }
        public Guid IdTenantAOO { get; set; }
        public bool IsRealResult { get; set; }

        #endregion

        #region [ Navigation Properties ]
        public ICollection<RoleModel> Children { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/Commons/RoleModel.ts
import CategoryFascicleRightModel = require("App/Models/Commons/CategoryFascicleRightModel");

interface RoleModel {
    UniqueId: string;
    EntityShortId: number;
    Name: string;
    IdTenantAOO: string;
    IdRole: number;
    FullIncrementalPath?: string;
    IsActive?: boolean;
    IdRoleFather?: number;
    ServiceCode: string;
    Children?: RoleModel[];
    CategoryFascicleRights: CategoryFascicleRightModel[];
    RoleTypology: string;
    IsRealResult: boolean;
}

export = RoleModel;